app/browser/desktopShare/chromeApi.js

```diff
@@ -1,27 +1,37 @@
+const { ipcRenderer } = require("electron")
+// In order to have this functionality working, contextIsolation should be disabled.
+// In new versions of electron, contextIsolation is set to true by default.
+// We should explicitly set it to false when creating BrowserWindow
+window.addEventListener("DOMContentLoaded", () => {
+  MediaDevices.prototype.getDisplayMedia = () => {
+    return new Promise((resolve, reject) => {
+      // Request main process to allow access to screen sharing
+      ipcRenderer.once('select-source', (event, source) => {
+        if (source) {
+          navigator.mediaDevices.getUserMedia({
```

Comment:
Can you put all the stuff from this if ina function? CodeBeat otherwise complains (as starts to be a bit deep and "complex")

Reply:
Sure, I'll make those changes? Do you know if CodeBeat vscode extension is present?

Reply:
not sure if there is a vscode extension. sorry.